packages/runtime-core/src/componentEmits.ts

```diff
@@ -152,6 +152,9 @@ export function emit(
   let handlerName
   let handler =
     props[(handlerName = toHandlerKey(event))] ||
+    // try kebab-case with on- (#9812)
```

Comment:
https://deploy-preview-9813--vue-sfc-playground.netlify.app/#eNqNUstuGzEM/BVBKCAbcHYB5+Zugj5goO2hDZreqh5cmeso0VKCpHUcGP73UFLsbF5GbiJnRA7J2fLPzlXrHviMN0F57SILEHt3LlF3zvrIvtrOsdbbjomqTkGiC4kS2x5V1BYZrAHjtwUuDfhRDsZsK5EVoIoLv4JYhXhnoFLWWM/OmFh5ABQfJe4kNnXpTV0piNA5s4hAEWNN7j+zeBJsBye54pnkw5aSP8OnRBCttULymoo09aAin/AYlMVWr6rrYJEGz1IlV9RIU7lfLg0VJJ+VIRK2MMbe/si56HuY7PPqCtTNK/nrsEk5yS88BPBrkPyAlXUUeH75Ezb0PoCdXfaG2EfA3xCs6ZPGQvvSP2zhwMtqv+fzaVz9CfNNBAz7oZLQxNxlvuR0zrTjt0Z/lHtaneZ/dDLa4t4KrziH1hsig05HOvQSWo0wpyCM/orHI4kJG0RT8W9MXijkC29dIlu8JMJ8jx/3yVKv82O7HX7bkVoao/nfx0hO/aSMVjfJP6Rn9FTNh+Jcyc9fgOOmLgVyh3eUmx6tN31WsKmL+IFR+e4ewO85Kg==


1. click `emit('some-event2')` button.
2. The browser will report an error

Reply:
In fact, it is not recommended to use `:on-some-event` to pass events. This will be passed to the component as `props`. The design of `vue` intentionally distinguishes `props` from `event` , that is, `props`, uses the `v-bind` directive, and `event` uses the `v-on` directive. I recommend you to use `v-on:some-event`

Reply:
Good catch, but that's another issue, when a string is passed through an event handler prop (no matter kebab-case or camelCase) [you can check in current version with camelCase](https://play.vuejs.org/#eNqNU01v2zAM/SuCMMAJEEtD0lPmFtmGANsOW7HuNu3g2kyi1vqAJKcZgvz3UnKaOmla9Gb6UeR7JN+WfraWrVugU1r4ykkbiIfQ2iuhpbLGBfLVKEsWziiSMR6DmJ4JLfSi1VWQRhNYgw7fSl034AYpGJKt0KQDWCjdEgLz4X8DrDKNceSSZEsHoLNPQu+ELnjXG7tiEEDZpgyAESFF6j81+sYomMd6l4L2Gwp6hI4RzhbGZIJyLFDwXjU6op2oXJWW3XmjUXYiKvaAF3TaUY//UGiMBV2FYP2U86rW+KyGRq4d0xC4torPMI27VgepIK+Nmk3YBfuYl41dleyC19KHPszAq/zWmQcPjllnaqwo6KjXkyOyBpc70DW4qPB9HE6eneNxknLEJdKILHAfO5xU8JXRC7k8mVOF65A49l82rv54XmXTmIcf6V9wLRw0VSuo7s/8v/ObTtu1g8SsN4fuaDp4fvMTNvh9AJWp22a/m1fA3+BN00aOXdqXdn8th7zE9ntau9TLP36+CaD9k6hINE0j5ae9xEt8Tfoz3Qmb9Kb4ZJgz/sLx+kBAyYB2qGEhNcwx8IO/mcdrztORZyPSi8bZvyE6pku+dsbG5P7xI/62m2q5Th/bbf/ZDtmijOK2DQH9PKsaWd1HnyGfwTGbD52/Bb16AQ4L3hVIHd5RbvxmvfFJwQJPOJLvWZruHgHfFJ87)
I know that is not recommended to use props to pass event handler, but as you can see in [docs](https://vuejs.org/guide/components/events.html#events-props), currently it's supported for camelCase

Reply:
Thanks for the clarification @pikax, the issue is when using `emit('event')` and the listener is defined as a hyphenated prop `:on-event` (the recommended case for props).

If it is not in the spec, I'm OK to close this PR and the issue, and add some clarification in docs about it.